packages/react/src/PageLayout/PageLayout.tsx

```diff
@@ -857,3 +910,75 @@ Header.__SLOT__ = Symbol('PageLayout.Header')
 Content.__SLOT__ = Symbol('PageLayout.Content')
 ;(Pane as WithSlotMarker<typeof Pane>).__SLOT__ = Symbol('PageLayout.Pane')
 Footer.__SLOT__ = Symbol('PageLayout.Footer')
+
+type RafAccumulatorOptions = {
+  immediate?: boolean
+}
+
+type RafAccumulatorControls = {
+  enqueue: (delta: number, options?: RafAccumulatorOptions) => void
+  flush: () => void
+  cancel: () => void
+}
+
+/**
+ * Batches numeric updates so that only a single callback runs per animation frame.
+ * Falls back to synchronous updates when the DOM is not available (SSR/tests).
+ */
+function useRafAccumulator(applyDelta: (delta: number) => void): RafAccumulatorControls {
```

Comment:
is this something we could extract from this component and maybe use as a pattern in other cases similar to this one?